color.h

```diff
@@ -14,6 +14,7 @@ class color
   /// @param a alpha value or opaqueness (0 = transparant, 255 = opaque)
   color(const int r = 255, const int g = 255, const int b = 255,
         const int a = 255);
+  std::string to_str() const noexcept;

```

Comment:
I think I understand why you think this is a good idea: other languages usually do this. In C++, however, [C.4: Make a function a member only if it needs direct access to the representation of a class](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-member). Or, in proper English, if something can be done by a (free (as in non-member)) function, do so. For a color, we can already get the R, G, B and alpha value and that is all the info we need to convert a color to a string.

Either make this a free function, or I will create a beginner Issue to do so.